mlir/lib/Dialect/SCF/Transforms/ForallToParallel.cpp

```diff
@@ -71,8 +71,9 @@ struct ForallToParallelLoop final
 
     parentOp->walk([&](scf::ForallOp forallOp) {
       if (failed(scf::forallToParallelLoop(rewriter, forallOp))) {
-        return signalPassFailure();
+        return WalkResult::skip();
```

Comment:
Is this a failure? Why remove the `signalPassFailure()`?
There is also likely a missing error message before `signalPassFailure()` here (we shouldn't fail silently).

Reply:
The result is not a real failure as it only occurs on match failure.

The main motivation for the change is that simply calling `signalPassFailure()` produces no output when the pass is called (at least from CLI). I'd expect the IR to remain unchanged in such case.
I think I should've captures the walk result and added some error on interruption. But there is no reason to interrupt on this error.

Perhaps a greedy rewriter could be better here instead of walking the graph manually.

Reply:
The `scf::forallToParallelLoop` function internally calls `notifyMatchFailure`, so some diagnostic should occur. That may not mean much if the pass terminates successfully though.

Reply:
`notifyMatchFailure` is a debug function. This is a question of semantics for the pass though, and unfortunately this pass does not even have a description! 
Can we start here and document the pass behavior before changing it?

Reply:
> This is a question of semantics for the pass though, and unfortunately this pass does not even have a description!

The lack of description is my fault. My original intention was to error out when `scf.forall` cannot be lowered. I don't think it makes sense to run this transform before bufferization, and after bufferization all `scf.forall` operations should produce no results.

Reply:
Good points all together. My change was too eager too.

@sabauma My view on the pass is that indicating full failure (through `signalPassFailure`) is a bit heavy handed in this case (and viewed it as "error") but if that is the intention, it is equally valid approach.
I'll leave the pass as is. Perhaps the description could be explicit about the intended behavior.